lib/bindings/c/src/lib.rs

```diff
@@ -971,22 +969,31 @@ pub async fn create_worker_selection_pipeline_chat(
         .component(component_name)?;
     let client = component.endpoint(GENERATE_ENDPOINT).client().await?;
 
-    let model_slug = Slug::from_string(model_name);
-    let card = match ModelDeploymentCard::load_from_store(
-        &Key::from_raw(model_slug.to_string()),
-        component.drt(),
-    )
-    .await
-    {
-        Ok(Some(card)) => card,
-        Ok(None) => anyhow::bail!("ModelDeploymentCard not found for model: {}", model_name),
-        Err(err) => anyhow::bail!(
-            "Error fetching ModelDeploymentCard from storage under key {model_slug}. {err}"
-        ),
-    };
+    // Discover the model card by searching all instances with this model name
+    use dynamo_llm::discovery::ModelWatcher;
+    let model_manager = std::sync::Arc::new(ModelManager::new());
+    let watcher = ModelWatcher::new(
+        component.drt().clone(),
+        model_manager.clone(),
+        router_mode,
+        kv_router_config,
+        busy_threshold,
+    );
+    let cards = watcher
+        .cards_for_model(model_name, Some(namespace), false)
+        .await
+        .with_context(|| format!("Failed to discover model: {}", model_name))?;
+
+    let mut card = cards.into_iter().next().ok_or_else(|| {
+        anyhow::anyhow!("ModelDeploymentCard not found for model: {}", model_name)
+    })?;
+
+    // Download tokenizer and model files from NATS to local disk
+    card.move_from_nats(distributed_runtime.nats_client())
+        .await
+        .with_context(|| "Failed to download model files from NATS")?;
```

Comment:
You don't need to do this any more. The files will be already available, because MX fetched them and put them in the PVC.